specs/ServerCertificate.md

```diff
@@ -0,0 +1,299 @@
+    e.Cancel = true;
+  }
+}
+
+// Function to validate the server certificate for untrusted root or self-signed certificate.
+bool ValidateServerCertificate(CoreWebView2Certificate certificate)
+{
+    // You may want to validate certificates in different ways depending on your app and scenario.
+    // One way might be the following:
+    // First, get the list of host app trusted certificates and its thumbprint.
+
+    // Then get the last chain element using `ICoreWebView2Certificate::get_PemEncodedIssuerCertificateChain`
+    // that contains the raw data of the untrusted root CA/self-signed certificate. Get the untrusted
+    // root CA/self signed certificate thumbprint from the raw certificate data and
+    // validate the thumbprint against the host app trusted certificate list.
+
+    // Finally, return true if it exists in the host app's certificate trusted list, or otherwise return false.
+    return true;
+}
+
+// This example clears the TLS decision in response to proceeding with TLS certificate errors.
+async void ClearServerCertificateErrorOverrideCache()
+{
+  bool isSuccessful = await webView.CoreWebView2.ClearServerCertificateErrorOverrideCacheAsync();
+  string message = (isSuccessful) ?
+                        "Clear server certificate error override cache succeeded." :
+                        "Clear server certificate error override cache failed.";
+  MessageBox.Show(this, "message", "Clear server certificate error override cache");
+}
+```
+
+# API Details
+
+## Win32 C++
+``` cpp
+[uuid(4B7FF0D2-8203-48B0-ACBF-ED9CFF82567A), object, pointer_default(unique)]
+interface ICoreWebView2_11 : ICoreWebView2_10 {
+  /// Add an event handler for the ServerCertificateError event.
+  /// The ServerCertificateError event is raised when the WebView2
+  /// cannot verify server's digital certificate while loading a web page.
+  ///
+  /// This event will raise for all web resources and follows the `WebResourceRequested` event.
+  ///
+  /// With this event you have several options for responding to server
+  /// certificate error requests:
+  ///
+  /// Scenario                                                   | Handled | Cancel
+  /// ---------------------------------------------------------- | ------- | ------
+  /// Ignore the warning and continue the request                | True    | False
+  /// Cancel the request                                         | n/a     | True
+  /// Display default TLS interstitial page                      | False   | False
+  ///
+  /// If you don't handle the event, WebView2 will show the default TLS interstitial page to user.
+  ///
+  /// WebView2 caches the response when `Handled` is `TRUE` for the RequestUri's host and the server certificate in the session
+  /// and the `ReceivingServerCertificateError` event won't be raised again for this host and certificate.
+  ///
+  /// To raise the event again you must clear the cache using `ClearServerCertificateErrorOverrideCache`.
```

Comment:
This name doesn't make sense to me currently, but need to look at it in light of any name changes discussed above to the event. The confusing thing is that there's no notion of caching in the event API, so this seems to be a reference to an implementation detail. If the event args get an enum, then this name could probably align with those.